File: MISC/MoveZeroes.cs
public class Solution {
    public void MoveZeroes(int[] nums) {
        // Shift non-zero values as far forward as possible
        // Fill the remaining space with zeros
        int pos = 0;
        for(int i=0; i<nums.Length; i++){
            if(nums[i] != 0){
                if(i != pos){
                    nums[pos] = nums[i];
                    nums[i] = 0;
                }
                pos ++;
            }
        }
    }
}

File: MISC/moveZeroes.py
class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        pos = 0
        for i in range(len(nums)):
            if nums[i] != 0:
                if i != pos:
                    nums[pos] = nums[i]
                    nums[i] = 0
                pos += 1